clipper_admin/test/clipper_manager_test.py

```diff
@@ -0,0 +1,280 @@
+        test_input = [99.3, 18.9, 67.2, 34.2]
+        req_json = json.dumps({'uid': 0, 'input': test_input})
+        headers = {'Content-type': 'application/json'}
+        response = requests.post(url, headers=headers, data=req_json)
+        parsed_response = json.loads(response.text)
+        self.assertNotEqual(parsed_response["output"], self.default_output)
+
+    def test_deployed_predict_function_queried_successfully(self):
+        model_version = 1
+        predict_func = lambda inputs: [str(len(x)) for x in inputs]
+        labels = ["test"]
+        input_type = "doubles"
+        result = self.clipper_inst.deploy_predict_function(
+            self.model_name_1, model_version, predict_func, labels, input_type)
+        self.assertTrue(result)
+
+        time.sleep(60)
+
+        received_non_default_prediction = False
+        url = "http://localhost:1337/{}/predict".format(self.app_name_1)
+        test_input = [101.1, 99.5, 107.2]
+        req_json = json.dumps({'uid': 0, 'input': test_input})
+        headers = {'Content-type': 'application/json'}
+        for i in range(0, 40):
+            response = requests.post(url, headers=headers, data=req_json)
+            parsed_response = json.loads(response.text)
+            output = parsed_response["output"]
+            if output == self.default_output:
+                time.sleep(20)
+            else:
+                received_non_default_prediction = True
+                self.assertEqual(int(output), len(test_input))
+                break
+
+        self.assertTrue(received_non_default_prediction)
+
+
+SHORT_ARGS = ['s', 'short']
+LONG_ARGS = ['l', 'long']
+ALL_ARGS = ['a', 'all']
+
+SHORT_TEST_ORDERING = [
+    'test_external_models_register_correctly',
+    'test_application_registers_correctly',
+    'get_app_info_for_registered_app_returns_info_dictionary',
+    'get_app_info_for_nonexistent_app_returns_none',
+    'test_add_container_for_external_model_fails',
+    'test_model_version_sets_correctly', 'test_get_logs_creates_log_files',
+    'test_inspect_instance_returns_json_dict',
+    'test_model_deploys_successfully',
+    'test_add_container_for_deployed_model_succeeds',
+    'test_predict_function_deploys_successfully'
+]
+
+LONG_TEST_ORDERING = [
+    'test_deployed_model_queried_successfully',
+    'test_deployed_predict_function_queried_successfully'
+]
+
+if __name__ == '__main__':
+    run_short = True
+    run_long = True
+    args_invalid = False
+    if len(sys.argv) > 1:
+        if sys.argv[1] in SHORT_ARGS:
+            run_long = False
+        elif sys.argv[1] in LONG_ARGS:
+            run_short = False
+        elif sys.argv[1] not in ALL_ARGS:
+            args_invalid = True
+    else:
+        args_invalid = True
+
+    if args_invalid:
+        print(
+            "Missing a parameter with value 's'/'short', 'l'/'long', or 'a'/'all' indicating which subset of tests should be executed!"
+        )
+        raise
```

Comment:
Used argparse. If no arguments are specified, all tests will execute (this is documented in the usage). Removed the exception.